config/initializers/unify_google_compute.rb

```diff
@@ -0,0 +1,207 @@
+require 'fog/google/models/compute/flavor'
+require 'fog/google/models/compute/image'
+require 'fog/google/models/compute/servers'
+require 'fog/google/models/compute/server'
+require 'fog/google/models/compute/disk'
+
+module Fog
+  module Compute
+    class Google
+      class Flavor
+        def vcpus
+          guest_cpus
+        end
+
+        def ram
+          memory_mb
+        end
+
+        def disk
+          image_space_gb || 0
+        end
+
+        def supported_architectures
+          'x86_64'
+        end
+
+        def id
+          name
+        end
+      end
+
+      class Image
+        def architecture
+          'x86_64'
+        end
+
+        alias_method :status_orig, :status
+        def status
+          s = status_orig
+
+          if s == 'READY'
+            'active'
+          else
+            s
+          end
+        end
+
+        def tags
+          {}
+        end
+
+        def id
+          name
+        end
+      end
+
+      class Servers
+        ZONE = 'us-central1-a'
+
+        alias_method :create_orig, :create
+        def create(params)
+          # Cyrrently hardcoded, in the feature it can be moved somewhere,
+          # for example into compute site configuration?
+          params[:zone] = ZONE
+
+          # Such user account with root privilage will be created on
+          # spawned machine. In the feature we can use user account here,
+          # but we need to answer the quesion what about production run, where
+          # machine can be shared amoud the users.
+          params[:username] = 'atmosphere'
+
+          disk = create_disk(params)
+          params[:disk] = disk.get_object(true, true,
+                                          'created_by_atmosphere', true)
+
+          server = nil
+          if params[:atmo_user_key]
+            key_file = Tempfile.open('userkey') do |f|
+              f.write(params[:atmo_user_key].public_key)
+              f
+            end
+            server = start_server(params.merge(key_path: key_file.path))
+          else
+            server = start_server(params)
+          end
+
+          if params[:user_data]
+            server.set_metadata(user_data: params[:user_data])
+          end
+
+          server
+        end
+
+        def destroy(id_at_site)
+          service.delete_server(id_at_site, ZONE)
+        end
+
+        private
+
+        def create_disk(params)
+          disk_params = {
+            name: params[:name],
+            # think about taking it from flavor or user requirement.
+            # Flavor disk is currently set to 0 so this also need to be
+            # rethinked.
+            size_gb: 10,
+            zone_name: params[:zone],
+            source_image: params[:image_id],
+            autoDelete: true
```

Comment:
I need to be camel case because it is converted into json body request send directly into google compute and google compute requires camel case.